### DISCUSSION TITLE
Array of objects from json file with objects

### DISCUSSION THREAD

**Discussion by Shuvi22** [nlohmann/json#3608] [Q&A] [answered]
I haven't quite figured out this library and I had a problem. I have a file with objects in it and john is fine accepting the string with them from the file, but doesn't do anything with it. How can I "unpack" this json into class objects? Are there any methods for this?

**Comment by falbrechtskirchinger** [comment, marked as answer]
Is this what you're looking for?
https://json.nlohmann.me/features/arbitrary_types/

**Reply by Shuvi22** [reply]
Unfortunately, not really. I have a string with an unknown number of objects and I want to make them all become objects of the c++ class. Here it says about a single object. Or I don't quite understand...

**Reply by falbrechtskirchinger** [reply]
Well, you need to start with a function for extracting or deserializing one object first. Then you can use `jsonArray.get<std::vector<MyClass>>()` to convert them all.

**Reply by falbrechtskirchinger** [reply]
Oh, wait. Can you clarify one thing:

Does your file contain multiple JSON objects?

```json
{}
{}
```

Or an array of objects?

```json
[
    {},
    {}
]
```

The first option can be a bit more difficult to deal with.

**Reply by Shuvi22** [reply]
`{
    "Items": [
        {},
        {}
    ]
}`

**Reply by falbrechtskirchinger** [reply]
```cpp
using nlohmann::json;

json j = json::parse(/*stream or string of JSON*/);
auto items = j.at("Items").get<std::vector<MyClass>>();
```

This assumes, that all objects map to the same class.

You still need to write a `void from_json(const json &j, MyClass &obj)` function as described in the link above. If your class is not default constructible, this method won't work – use [this](https://json.nlohmann.me/features/arbitrary_types/#how-can-i-use-get-for-non-default-constructiblenon-copyable-types) instead.

**Reply by Shuvi22** [reply]
I'm going to use this a little later. Have a great day and thanks for your help!
